tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -80,14 +83,8 @@ def rewrite_zip(root_dir, zip_filepath):
       zip_file.extractall(path=temp_dir)
 
     rewrite_dir(temp_dir)  # Recursivelly compress the archive content
-
-    # Compressed the .zip file again
-    with zipfile.ZipFile(zip_filepath, 'w') as zip_file:
-      for file_dir, _, files in os.walk(temp_dir):
-        for file in files:
-          file_path = os.path.join(file_dir, file)
-          zip_file.write(file_path)
-
+    shutil.make_archive(zip_filepath[:-4], 'zip', temp_dir) #Copies the directory structure of the extracted file ans stores it back as zip
```

Comment:
`zip_filepath[:len(.zip)]`, also I would add a check to make sure the file ends by `.zip`

Reply:
In rewrite_dir the zipfile.is_zipfile(path) check is done. To the best of my knowledge, I know only .zip extensions which would be allowed by this filter. Do you know of other '.zip' extensions which come to mind that might create issues ?
In any case, in the latest commit, I perform a possibly redundant check just as you mentioned above.